-1

**Failed Tests**:   Build  HeaderConsistency  ClangBuild 
**Size**: This PR adds an extra 24KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dfba33/41438/summary.html
**COMMIT**: 0fa70c1c2b84551e140eb98f55bfeb4fd0b7e601
**CMSSW**: CMSSW_14_2_X_2024-09-09-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/45965/41438/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @Dr15Jones cms-sw/cmssw#45745
 - @mmusich cms-sw/cmssw#45954
 - @smuzaffar cms-sw/cmssw#45916

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dfba33/41438/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dfba33/41438/git-merge-result


## Build

I found compilation error when building: 

<pre>>> Compiling  src/OnlineDB/CSCCondDB/src/CSCMap.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_14_2_X_2024-09-09-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_14_2_X_2024-09-09-2300' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-09-09-2300/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/pcre/8.43-e34796d17981e9b6d174328c69446455/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/boost/1.80.0-7ac88d918a6883d593abc133ee0eb2e9/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/clhep/2.4.7.1-17c7283747b6cbc68a3534cb987a890c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/gsl/2.6-4e597de26ced42091883404fa44cef2e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/oracle/19.11.0.0.0dbru-0ef9ee763c1e7a90d8c4515a5af97f0b/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/lcg/root/6.30.09-5371975256c236eab45164692fccb0e2/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/tbb/v2021.9.0-2391c941213c757dc9a1835b31681235/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/zlib/1.2.11-1a082fc322b0051b504cc023f21df178/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-39786ff94f8aa2f543922cad720e1b32/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-39786ff94f8aa2f543922cad720e1b32/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/fmt/10.2.1-deba157dd76fed17ac8b42adc60ff53f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/OpenBLAS/0.3.27-6574458f8642b299ac965c654e194a8b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/tinyxml2/6.2.0-68d511689c5dbd9a0d296bbb795fa052/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v2 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/OnlineDB/CSCCondDB/src/OnlineDBCSCCondDB/CSCMap.cc.d src/OnlineDB/CSCCondDB/src/CSCMap.cc -o tmp/el8_amd64_gcc12/src/OnlineDB/CSCCondDB/src/OnlineDBCSCCondDB/CSCMap.cc.o
>> Compiling  src/OnlineDB/CSCCondDB/src/CSCMap1.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_14_2_X_2024-09-09-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_14_2_X_2024-09-09-2300' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-09-09-2300/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/pcre/8.43-e34796d17981e9b6d174328c69446455/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/boost/1.80.0-7ac88d918a6883d593abc133ee0eb2e9/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/clhep/2.4.7.1-17c7283747b6cbc68a3534cb987a890c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/gsl/2.6-4e597de26ced42091883404fa44cef2e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/oracle/19.11.0.0.0dbru-0ef9ee763c1e7a90d8c4515a5af97f0b/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/lcg/root/6.30.09-5371975256c236eab45164692fccb0e2/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/tbb/v2021.9.0-2391c941213c757dc9a1835b31681235/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/zlib/1.2.11-1a082fc322b0051b504cc023f21df178/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-39786ff94f8aa2f543922cad720e1b32/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-39786ff94f8aa2f543922cad720e1b32/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/fmt/10.2.1-deba157dd76fed17ac8b42adc60ff53f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/OpenBLAS/0.3.27-6574458f8642b299ac965c654e194a8b/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02854/el8_amd64_gcc12/external/tinyxml2/6.2.0-68d511689c5dbd9a0d296bbb795fa052/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v2 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/OnlineDB/CSCCondDB/src/OnlineDBCSCCondDB/CSCMap1.cc.d src/OnlineDB/CSCCondDB/src/CSCMap1.cc -o tmp/el8_amd64_gcc12/src/OnlineDB/CSCCondDB/src/OnlineDBCSCCondDB/CSCMap1.cc.o
In file included from src/OnlineDB/CSCCondDB/src/CSCFitAFEBThr.cc:5:
src/OnlineDB/CSCCondDB/interface/CSCThrTurnOnFcn.h:14:38: error: missing binary operator before token "("
   14 | #if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
      |                                      ^
src/OnlineDB/CSCCondDB/interface/CSCThrTurnOnFcn.h:45:38: error: missing binary operator before token "("
   45 | #if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
      |                                      ^
</pre>

## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Entering Package Validation/RecoVertex
>> Entering Package Validation/SiTrackerPhase2V
>> Entering Package Validation/TrackerRecHits
>> Entering Package Validation/TrackingMCTruth
>> Compile sequence completed for CMSSW CMSSW_14_2_X_2024-09-09-2300
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
+ eval scram build outputlog '&&' '(python3' /data/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --logDir /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_2_X_2024-09-09-2300/tmp/el8_amd64_gcc12/cache/log/src '||' 'true)'
++ scram build outputlog
>> Entering Package CondCore/SiStripPlugins
Entering library rule at src/CondCore/SiStripPlugins/plugins
>> Compiling edm plugin src/CondCore/SiStripPlugins/plugins/plugin.cc
</pre>

--- a/OnlineDB/CSCCondDB/interface/CSCThrTurnOnFcn.h
+++ b/OnlineDB/CSCCondDB/interface/CSCThrTurnOnFcn.h
@@ -11,6 +11,10 @@
 
 #include "Minuit2/FCNBase.h"
 #include <vector>
+#include <RVersion.h>
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
+#include <span>
+#endif
 
 class CSCThrTurnOnFcn : public ROOT::Minuit2::FCNBase {
 private:
@@ -39,7 +43,11 @@ public:
   void setNorm(float n) { norm = n; };
 
   /// Provide the chi-squared function for the given data
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
+  double operator()(std::span<const double>) const override;
+#else
   double operator()(const std::vector<double>&) const override;
+#endif
 
   ///@@ What?
   double Up() const override { return 1.; }

--- a/OnlineDB/CSCCondDB/src/CSCFitAFEBThr.cc
+++ b/OnlineDB/CSCCondDB/src/CSCFitAFEBThr.cc
@@ -130,8 +130,12 @@ bool CSCFitAFEBThr::ThresholdNoise(const std::vector<float>& inputx,
   //                                               <<" "<<ery[i]<<std::endl;
 
   /// Fit  as 1D, <=500 iterations, edm=10**-5 (->0.1)
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
+  FunctionMinimum fmin =
+      theFitter->Minimize(*theOBJfun, std::span<double>(parinit), std::span<double>(erparinit), 1, 500, 0.1);
+#else
   FunctionMinimum fmin = theFitter->Minimize(*theOBJfun, parinit, erparinit, 1, 500, 0.1);
-
+#endif
   status = fmin.IsValid();
 
   if (status) {

--- a/OnlineDB/CSCCondDB/src/CSCThrTurnOnFcn.cc
+++ b/OnlineDB/CSCCondDB/src/CSCThrTurnOnFcn.cc
@@ -5,7 +5,11 @@
 #include <vector>
 #include "TMath.h"
 
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
+double CSCThrTurnOnFcn::operator()(std::span<const double> par) const {
+#else
 double CSCThrTurnOnFcn::operator()(const std::vector<double>& par) const {
+#endif
   double x, y, er, fn;
   double N = norm;
   double chi2 = 0.;

--- a/RecoVertex/BeamSpotProducer/interface/BSpdfsFcn.h
+++ b/RecoVertex/BeamSpotProducer/interface/BSpdfsFcn.h
@@ -18,6 +18,10 @@ ________________________________________________________________**/
 
 #include <iostream>
 #include <string>
+#include <RVersion.h>
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 32, 4)
+#include <span>
+#endif
 
 class BSpdfsFcn : public ROOT::Minuit2::FCNBase {
 public:
@@ -26,14 +30,25 @@ public:
   // define pdfs to use
   void SetPDFs(std::string usepdfs) { fusepdfs = usepdfs; }
 
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
+  double operator()(std::span<const double>) const override;
+#else
   double operator()(const std::vector<double>&) const override;
+#endif
   double Up() const override { return 1.; }
 
 private:
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
+  double PDFGauss_d(double z, double d, double sigmad, double phi, std::span<const double> parms) const;
+  double PDFGauss_d_resolution(double z, double d, double phi, double pt, std::span<const double> parms) const;
+
+  double PDFGauss_z(double z, double sigmaz, std::span<const double> parms) const;
+#else
   double PDFGauss_d(double z, double d, double sigmad, double phi, const std::vector<double>& parms) const;
   double PDFGauss_d_resolution(double z, double d, double phi, double pt, const std::vector<double>& parms) const;
 
   double PDFGauss_z(double z, double sigmaz, const std::vector<double>& parms) const;
+#endif
 
   std::string fusepdfs;
   std::vector<BSTrkParameters> fBSvector;

--- a/RecoVertex/BeamSpotProducer/interface/FcnBeamSpotFitPV.h
+++ b/RecoVertex/BeamSpotProducer/interface/FcnBeamSpotFitPV.h
@@ -18,6 +18,8 @@
 #include "Minuit2/FCNBase.h"
 
 #include <vector>
+#include <span>
+#include <RVersion.h>
 
 class FcnBeamSpotFitPV : public ROOT::Minuit2::FCNBase {
 public:
@@ -29,7 +31,11 @@ public:
   // deltaFcn for definition of the uncertainty
   double Up() const override { return errorDef_; }
   // -2lnL value based on vector of parameters
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 32, 4)
+  double operator()(std::span<const double>) const override;
+#else
   double operator()(const std::vector<double>&) const override;
+#endif
   // vertex count used for the fit (after selection)
   unsigned int nrOfVerticesUsed() const;
 

--- a/RecoVertex/BeamSpotProducer/src/BSpdfsFcn.cc
+++ b/RecoVertex/BeamSpotProducer/src/BSpdfsFcn.cc
@@ -15,9 +15,16 @@ ________________________________________________________________**/
 
 #include <cmath>
 #include <vector>
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
+#include <span>
+#endif
 
 //______________________________________________________________________
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 32, 4)
+double BSpdfsFcn::PDFGauss_d(double z, double d, double sigmad, double phi, std::span<const double> parms) const {
+#else
 double BSpdfsFcn::PDFGauss_d(double z, double d, double sigmad, double phi, const std::vector<double>& parms) const {
+#endif
   //---------------------------------------------------------------------------
   //  PDF for d0 distribution. This PDF is a simple gaussian in the
   //  beam reference frame.
@@ -35,8 +42,12 @@ double BSpdfsFcn::PDFGauss_d(double z, double d, double sigmad, double phi, cons
 }
 
 //______________________________________________________________________
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 32, 4)
+double BSpdfsFcn::PDFGauss_d_resolution(double z, double d, double phi, double pt, std::span<const double> parms) const {
+#else
 double BSpdfsFcn::PDFGauss_d_resolution(
     double z, double d, double phi, double pt, const std::vector<double>& parms) const {
+#endif
   //---------------------------------------------------------------------------
   //  PDF for d0 distribution. This PDF is a simple gaussian in the
   //  beam reference frame. The IP resolution is parametrize by a linear
@@ -57,7 +68,12 @@ double BSpdfsFcn::PDFGauss_d_resolution(
 }
 
 //______________________________________________________________________
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 32, 4)
+double BSpdfsFcn::PDFGauss_z(double z, double sigmaz, std::span<const double> parms) const {
+#else
 double BSpdfsFcn::PDFGauss_z(double z, double sigmaz, const std::vector<double>& parms) const {
+#endif
+
   //---------------------------------------------------------------------------
   //  PDF for z-vertex distribution. This distribution
   // is parametrized by a simple normalized gaussian distribution.
@@ -73,7 +89,11 @@ double BSpdfsFcn::PDFGauss_z(double z, double sigmaz, const std::vector<double>&
 }
 
 //______________________________________________________________________
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 32, 4)
+double BSpdfsFcn::operator()(const std::span<const double> params) const {
+#else
 double BSpdfsFcn::operator()(const std::vector<double>& params) const {
+#endif
   double f = 0.0;
 
   //std::cout << "fusepdfs=" << fusepdfs << " params.size="<<params.size() << std::endl;

--- a/RecoVertex/BeamSpotProducer/src/FcnBeamSpotFitPV.cc
+++ b/RecoVertex/BeamSpotProducer/src/FcnBeamSpotFitPV.cc
@@ -51,7 +51,11 @@ unsigned int FcnBeamSpotFitPV::nrOfVerticesUsed() const {
   return nVtx;
 }
 
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 33, 1)
+double FcnBeamSpotFitPV::operator()(std::span<const double> pars) const {
+#else
 double FcnBeamSpotFitPV::operator()(const std::vector<double>& pars) const {
+#endif
   //
   // fit parameters
   //

